### ISSUE TITLE
New pluginloader fails to generate plugin type string from class name upon Node construction

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#598]
When creating an instance of a calculation from a plugin that is loaded through the new pluginloader, the `entry_point_tpstr_from` method from the pluginloader is supposed to generate the type string based on the class name of the calculation, to be stored in the type column of the node table. However, currently this fails. I think this is because of an error in `get_class_to_entry_point_map` in `aiida.common.pluginloader`. The following call on line 217 is the problem I think:

```
groups = (g for g in epm.get_entry_map('aiida-core').iterkeys() if g.startswith('aiida'))
```

Now `groups` will contain only one entry `aiida-core` since that is the top level of the entry map. When it then proceeds to iterate over it, it does not actually access the groups within it. I think this is simply missing one level of dereferencing before iterating over the groups, i.e. by doing the following:

```
groups = (g for g in epm.get_entry_map('aiida-core')['aiida-core'].iterkeys() if g.startswith('aiida'))
```

This actually solves the issue for me, as now the correct type string is found and returned, however I am not sure if this is truly solving the underlying problem or if this is just monkeypatching something.

**Comment by greschd** [comment]
See https://github.com/DropD/reentry/pull/8

**Comment by giovannipizzi** [comment]
Is the new version of reentry already out? 
On pypi it looks it's 1.0.1. Is this working properly? If so, maybe you can make a PR to fix the dependencies of AiiDA to reentry>=1.0.1 or something similar?

**Comment by greschd** [comment]
No, 1.0.1 is the old version. In principle updating should be a pretty simple thing to do, but we need @DropD to do it (since only he has access). If you really want, you could instead change the dependency to point to the git master (or a specific commit).

**Comment by DropD** [comment]
@greschd So, your fix in reentry should fix this issue too, right?

In general you can update by simply pulling the fixed version repo, pip
uninstall reentry, pip install -e reentry in your virtualenv.

I will release the new version today.

Am 20.06.2017 10:43 schrieb "Dominik Gresch" <notifications@github.com>:

No, 1.0.1 is the old version. In principle updating should be a pretty
simple thing to do, but we need @DropD <https://github.com/dropd> to do it
(since only he has access). If you really want, you could instead change
the dependency to point to the git master (or a specific commit).

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/aiidateam/aiida_core/issues/598#issuecomment-309686116>,
or mute the thread

**Comment by greschd** [comment]
Yes, that should fix it.

**Comment by DropD** [comment]
DropD/reentry has released [version 1.0.2](https://pypi.python.org/pypi/reentry/1.0.2) with this fix in it, if you can confirm that typestrings get generated correctly now, we can close this

**Comment by sphuber** [comment]
That seems to have fixed it, after running `reentry scan` following the installation of `1.0.2`.
Question: should we now require `reentry>=1.0.2` in setup scripts of aiida-core? Currently it is set to `reentry>=0.9.1`

**Comment by greschd** [comment]
Yes, I think that makes sense.

**Comment by sphuber** [comment]
OK I will create the PR
